Storage/SQL/SQLiteLogins.swift

```diff
@@ -212,7 +220,9 @@ public class SQLiteLogins: BrowserLogins {
         "\(TableLoginsMirror) WHERE is_overridden = 0 AND hostname IS ? " +
         "ORDER BY timeLastUsed DESC"
 
-        let args: Args = [protectionSpace.host, protectionSpace.host]
+        // Since we store hostnames as the full scheme/protocol + host, combine the two to look up in our DB.
+        let hostname = "\(protectionSpace.`protocol`!)://\(protectionSpace.host)"
```

Comment:
Add an extension method on `NSURLProtectionSpace` to return `https://foo.com:8080`. Call through to that from `Login.hostname`.